src/schoology-extractor/schoology_extractor/api/request_client.py

```diff
@@ -8,35 +8,53 @@
 
 from requests_oauthlib import OAuth1Session
 
+from .paginated_result import PaginatedResult
+
 
 DEFAULT_URL = "https://api.schoology.com/v1/"
 
 
 class RequestClient:
-    """
-    The RequestClient class wraps all the configuration complexity related to authentication and http requests for Schoology API
+    '''
+    The RequestClient class wraps all the configuration complexity related
+    to authentication and http requests for Schoology API
+
+    Parameters
+    ----------
+    schoology_key : str
+        The consumer key given by Schoology.
+    schoology_secret : str
+        The consumer secret given by Schoology.
+    base_url : Optional[str]
+        The API base url. Default value: https://api.schoology.com/v1/
 
-    Args:
-        schoology_key (str): The consumer key given by Schoology
-        schoology_secret (str): The consumer secret given by Schoology
-        base_url (str, optional): The API base url. Default value: https://api.schoology.com/v1/
+    Attributes
+    ----------
+    oauth : OAuth1Session
+        The two-legged authenticated OAuth1 session.
+    '''
```

Comment:
I had not previously noticed the ability to describe attributes like this. Interesting.